### PR TITLE
Adgudime/apex sac

### PR DESCRIPTION
**Pull request by AdityaGudimella**
## Why are these changes needed?

The config for ApexSAC incorrectly had `target_network_update` set to a non-zero value. I fixed that, and updated the ApexSAC CartPole-v1 test to indicate that it now learns on CartPole-v1

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: rllib/tests/agents/test_learning.py [hunk 1]
```diff
@@ -80,6 +80,7 @@ def test_monotonically_improving_algorithms_can_converge_with_different_framewor
     for i in range(n_iter):
         results = trainer.train()
         episode_reward_mean = results["episode_reward_mean"]
+        print(episode_reward_mean)
```
Comment (Edilmo) [inline rllib/tests/agents/test_learning.py:83]: can we remove this?

File: rllib/agents/sac/apex.py [hunk 2]
```diff
@@ -1,35 +1,35 @@
...
+        },
         "n_step": 1,
         "num_gpus": 0,
         "num_workers": 32,
         "buffer_size": 2000000,
         "learning_starts": 50000,
         "train_batch_size": 512,
         "rollout_fragment_length": 50,
-        "target_network_update_freq": 500000,
+        "target_network_update_freq": 0,
```
Comment (Edilmo) [inline rllib/agents/sac/apex.py:24]: This implies updating the target model at every step, isn't it?
Just curious, why in SAC we want to do that?

Comment (AdityaGudimella) [reply rllib/agents/sac/apex.py:24]: Yes. This is the default setting for SAC. For Apex the default setting is 500,000. SAC just uses the target network a bit differently than the other algorithms. In other algorithms, the target network is meant to stabilize the Q Net updates. SAC instead uses twin q networks to do that. We might want to set this value higher if we aren't using twin q networks in SAC.

File: rllib/tests/agents/test_learning.py (hunk 1 above)
Comment (AdityaGudimella) [reply rllib/tests/agents/test_learning.py:83]: Sorry. Forgot about this. Will remove.

Comment (AdityaGudimella) [reply rllib/tests/agents/test_learning.py]: Done.

File: rllib/agents/sac/apex.py (hunk 2 above)
Comment (Edilmo) [reply rllib/agents/sac/apex.py:24]: Got it

**Pull request merged**